### ISSUE TITLE
🚀[2020] Fine-Tuning Pretrained Language Models: Weight Initializations, Data Orders, and Early Stopping

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#43]
### タイトル
Fine-Tuning Pretrained Language Models: Weight Initializations, Data Orders, and Early Stopping
### リンク
https://arxiv.org/abs/2002.06305
### 著者
Jesse Dodge, Gabriel Ilharco, Roy Schwartz, Ali Farhadi, Hannaneh Hajishirzi, Noah Smith
### 投稿日付
(Submitted on 15 Feb 2020)
## 1. どんなもの？

BERT などの pretrain モデルを finetune する際の学習のダイナミクスに関する研究。GLUE を構成するタスクの一部 (MRPC, RTE, CoLA, SST) について、モデルや pretrained weight, ハイパーパラメータ, 学習方法は同じまま seed だけを変えて 2,100 回の finetune を行い、性質や精度を上げるための方法について議論している。

## 2. 先行研究と比べてどこがすごい？

次の2つの random 性のある部分を区別して seed を変えて実験している。

- weight initialization
- training data order

## 3. 技術や手法のキモはどこ？

BERT について様々な seed で finetune の実験をしたところ、次のような発見が得られた。

- 一部タスクで BERT 提案モデルの精度どころか XLNet, RoBERTa, ALBERT を上回った
- RTE と CoLA タスクにおいては既存スコアを(絶対量で)7%も上回る精度を得られた
- 今回実験したすべてのタスクに対して有効となる weight initialization の seed を発見した

また、今後の学習ダイナミクスに関する研究のため、2,100回分の学習に関するデータを全て公開しているとのこと(まだ?見つからない)。

## 5. 議論はある？

- 他の pretrain モデルでも同様か (XLNet, RoBERTa, ALBERT, etc.)
- Binary Classification 以外のタスクでも同様か
- データ数の多い finetune タスクでも同様か (SST では variance が明らかに小さくなっている)

**Comment by cfiken** [comment]
## 7. コメント

- seed で7%とか違うのはデータ量の少ないタスク(下記表)
    - ![スクリーンショット 2020-03-24 00 03 02](https://user-images.githubusercontent.com/1255485/77330533-d5d5c280-6d62-11ea-8422-657e383a54b5.png)
    - 見て分かる通り、validation データ数は SST 以外は数百~1000しかない。
    - seed で精度が変わるのは事実だが、7% も変わるのはデータセットの特性によるものが大きい気がするがどうだろう...
        - 実際 SST ではかなり分散が小さくなっている
- weight initialization で4つのタスク全てで良い seed が見つかったのは新たな initialization 手法の機運...?
